### ISSUE TITLE
snapshot_copy does not work as per example

### ISSUE THREAD

**Issue by andybusuu** [terraform-aws-modules/terraform-aws-redshift#73]
## Description

The `snapshot_copy` functionality does not work as per [examples/complete/main.tf](https://github.com/terraform-aws-modules/terraform-aws-redshift/blob/v4.0.2/examples/complete/main.tf) 

This does NOT work:
```terraform
  snapshot_copy = {
    useast1 = {
      destination_region = "us-east-1"
      grant_name         = aws_redshift_snapshot_copy_grant.useast1.snapshot_copy_grant_name
    }
  }
```

This works:
```terraform
  snapshot_copy = {
    destination_region = "us-east-1"
    grant_name         = aws_redshift_snapshot_copy_grant.useast1.snapshot_copy_grant_name
    retention_period   = 1
  }
```

Noticed this when upgrading from `v3.4.0` to `v4.0.2`
## Versions

- Module version [Required]: `v4.0.2`

- Terraform version:
`Terraform v1.0.11`
- Provider version(s):
```
+ provider registry.terraform.io/hashicorp/aws v4.36.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
```

## Expected behavior

Snapshot copies to be configured

## Actual behavior

Existing snapshot copies are removed.

### Terminal Output Screenshot(s)

![image](https://user-images.githubusercontent.com/98590547/198342250-5c738ab4-73ac-4c59-9365-78d73f517a24.png)

**Comment by kchitturi001** [comment]
Thank you for the post. was wasting my time on this for some time.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 5.0.0 :tada:
